src/ui/StatusEditor.svelte

```diff
@@ -11,6 +11,36 @@
 
     let statusSymbol = task.status.symbol;
 
+    /**
+     * Set the done or cancelled date field of the editable task.
+     * These fields are connected with the status.
+     *
+     * The date should be set in either of 2 cases:
+     * - the date field is empty and the status was set (set the date from the task with the applied status)
+     * - the date field is not empty but another status was set (clean the date field)
```

Comment:
It's a little bit of a smell when a private function needs this much docs! 😄 

I would move lines 19 and 20 to the bits of code that they are referring to (possibly 34.5 and 39.5) as I still don't find the logic or behaviour of the function to be immediately obvious.

Reply:
> I would move lines 19 and 20 to the bits of code that they are referring to (possibly 34.5 and 39.5) as I still don't find the logic or behaviour of the function to be immediately obvious.

This is done. Resolving